pyproject.toml

```diff
@@ -43,27 +44,9 @@ nbformat = "^5.10"
 ipywidgets = "^8.1.3"
 
 ######################  The Tensorflow Section ######################
-# Dedicated for making sure that poetry can install tensorflow on all platforms.
-# This hopefully is a temporary workaround, pending TF starting doing a good job with their pypi release metadata.
-
-# Issue between poetry and tensorflow metadata since >=2.11
-# related to https://github.com/python-poetry/poetry/issues/8271
-# Inspired from https://github.com/tensorflow/tensorflow/blob/adb39b04e9cb116df4659a7e2de9eea27e62f25c/tensorflow/tools/pip_package/setup.py#L148-L162
-tensorflow = {version = "^2.15.0" }
-tensorflow-macos = { version = "2.15.0", platform = "darwin", markers = "platform_machine=='arm64'" }
-tensorflow-intel = { version = "^2.15.0", platform = "win32" }
-# Disabled to prevent taking over GPU:
-# tensorflow-cpu = [
-#     { version = "^2.15.0", platform = "linux", markers = "platform_machine!='arm64' and platform_machine!='aarch64'" },
-#     { version = "^2.15.0", platform = "darwin", markers = "platform_machine!='arm64' and platform_machine!='aarch64'" },]
-tensorflow-cpu-aws = { version = "^2.15.0", platform = "linux", markers = "platform_machine=='arm64' or platform_machine=='aarch64'" }
-# https://github.com/tensorflow/tensorflow/blob/adb39b04e9cb116df4659a7e2de9eea27e62f25c/tensorflow/tools/pip_package/setup.py#L107-L108
-# https://github.com/python-poetry/poetry/issues/8271#issuecomment-1697740447
-tensorflow-io-gcs-filesystem = [
-    { version = ">= 0.23.1", markers = "platform_machine!='arm64' or platform_system!='Darwin'" },
-    { version = "< 0.32.0", markers = "platform_system == 'Windows'" }
-]
-tensorflow-probability = { version = ">=0.22.0,<1.0", extras = ["tf"] }
+tensorflow = {version = "^2.15.0, <2.19.0" }
```

Comment:
is 2.15 actually supported now with all these changes? wondering if we should add a test suite that manually downgrades to 2.15 and confirms this. i know it's not _Poetic_, but I suspect it's the most likely lower bound dependency that will suddenly fail.

Reply:
For python versions, it's fairly easy. Just use the `strategy.matrix` property like we do here:
https://github.com/XanaduAI/MrMustard/blob/2a1d7f97526bb36fbbff90b22ee686b001c122c0/.github/workflows/builds.yml#L15-L18
Then pass the `${{ matrix.python-version }}` value to `actions/setup-python`.

Poetry won't help with testing various tensorflow versions 😢 We'd have to work against the grain and manually pip install versions other than the one specific in `poetry.lock`. It's hacky, but can be done.

If we think either/both of these things are valuable, I can add it.

Reply:
I think it's worthwhile to have. My only concern is slowing down our CICD so the parameterizations should run in parallel. For now lets just stick to running over the different versions of python 👍